atest/robot/core/filter_by_names.robot

```diff
@@ -157,6 +157,8 @@ Filter Using Suite Long Name
     Run Suites    --suite Subsuites.Sub1
     Should Contain Suites    ${SUITE}    Subsuites
     Should Contain Tests   ${SUITE}   SubSuite1 First
+    Run Suites    --suite suite_should.be.parsed  ${DATA DIR}/misc
+    Stderr Should Be Empty
 
```

Comment:
It would be good to test that the right tests, and only them, have been actually run. Should be pretty easy with `Should Contain Suites` that's used already above.

Related to this, it would be better to have different names in tests in the new `should/be/parsed.robot` and `should/not be/parsed.robot` files to make it explicit which test we expect to be executed. A name like `A test in 'should/be/parsed.robot'` would be extremely explicit.